src/main/java/org/sonarsource/sonarlint/ls/connected/SecurityHotspotsApiServer.java

```diff
@@ -0,0 +1,197 @@
+ */
+package org.sonarsource.sonarlint.ls.connected;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.apache.hc.core5.http.ClassicHttpRequest;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.Header;
+import org.apache.hc.core5.http.HttpException;
+import org.apache.hc.core5.http.NameValuePair;
+import org.apache.hc.core5.http.impl.bootstrap.HttpServer;
+import org.apache.hc.core5.http.impl.bootstrap.ServerBootstrap;
+import org.apache.hc.core5.http.io.HttpFilterChain;
+import org.apache.hc.core5.http.io.HttpFilterHandler;
+import org.apache.hc.core5.http.io.HttpRequestHandler;
+import org.apache.hc.core5.http.io.SocketConfig;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.protocol.HttpContext;
+import org.apache.hc.core5.io.CloseMode;
+import org.apache.hc.core5.net.URLEncodedUtils;
+import org.sonarsource.sonarlint.core.client.api.common.LogOutput;
+import org.sonarsource.sonarlint.ls.log.LanguageClientLogOutput;
+
+public class SecurityHotspotsApiServer {
+
+  static final int STARTING_PORT = 64120;
+  static final int ENDING_PORT = 64130;
+
+  private static final int INVALID_PORT = -1;
+
+  private final LanguageClientLogOutput output;
+
+  private HttpServer server;
+  private int port;
+
+  public SecurityHotspotsApiServer(LanguageClientLogOutput output) {
+    this.output = output;
+  }
+
+  public void init(String ideName, String clientVersion, @Nullable String workspaceName) {
+    final SocketConfig socketConfig = SocketConfig.custom()
+      .setSoTimeout(15, TimeUnit.SECONDS)
+      .setTcpNoDelay(true)
+      .build();
+    port = INVALID_PORT;
+    int triedPort = STARTING_PORT;
+    HttpServer startedServer = null;
+    while(port < 0 && triedPort <= ENDING_PORT) {
+      try {
+        startedServer = ServerBootstrap.bootstrap()
+          .setListenerPort(triedPort)
+          .setSocketConfig(socketConfig)
+          .addFilterFirst("CORS", new CorsFilter())
+          .register("/sonarlint/api/status", new StatusRequestHandler(ideName, clientVersion, workspaceName))
+          .register("/sonarlint/api/hotspots/show", new ShowHotspotRequestHandler(output))
+          .create();
+        startedServer.start();
+        port = triedPort;
+      } catch (Exception t) {
+        output.log("Error while starting port: " + t.getMessage(), LogOutput.Level.ERROR);
+        triedPort++;
+      }
+    }
+    if (port > 0) {
+      output.log("Started hotspots server on port " + port, LogOutput.Level.INFO);
```

Comment:
I wonder if we should not try to find a better name than `hotspots server` for this component. It might be misleading for end users...

Reply:
Maybe "security hotspot handler"?